### DISCUSSION TITLE
Signed commits and tags

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#2096] [Ideas]
When we now start to tag our own releases, I would like to require at least the tags to be signed.
https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work

Also bigger imports or patches would be great, if the commit itself was signed.

How is the opinion about this here?

**Comment by colinmollenhour** [comment]
How is this enforced? Also how does this work with github's web-based "Merge" button?

**Comment by Flyingmana** [comment]
There is no enforcement (yet)
In a far distant future tools like composer could check if a tag was signed and are able to deny install for unsigned ones.

as the web ui just adds a merge commit, it does not interfere if the commits inside this PR are signed.

For Tags, we could also sign the downloads, this would be a manual process which is I think currently useless, as for example composer would not make use of it

**Comment by robbieaverill** [comment]
Signing tags makes sense since they are a point-in-time representation of a set of changes. Signing everything doesn't make as much sense. Since you can't tag via GitHub (AFAIK?) there's no downside to enforcing this as policy when releasing new tags.

**Comment by drobinson** [comment]
Generally I'm the one doing tagging, so please help me remember to do this next time we create a release and I'll tag it.  Is it possible to sign old tags somehow @Flyingmana ?

**Comment by seansan** [comment]
Can we add tis to the readme? Or is there a better place? and close the issue

**Comment by sreichel** [comment]
> Can we add tis to the readme? Or is there a better place? and close the issue

This is something cor contribution guidelines  .... working on it.
